src/api_manager/check_workflow.cc

```diff
@@ -39,6 +40,8 @@ void CheckWorkflow::RegisterAll() {
   Register(CheckServiceControl);
   // Quota control
   Register(QuotaControl);
+  // Update API base path
+  Register(ApiBasePath);
```

Comment:
For such operation,  I don't think we need to register this as "a control".  Usually if "a control" is async which is making a remote call, and waiting a callback to continue, we do it this  way.
StripBathPath is not "async call";

Reply:
As the order,   if we do the basepath manipulation as the first thing in our filter,  we don't event need to change path_match change.